File: keep-multiplying-found-values-by-two.ts
{


    function findFinalValue(nums: number[], original: number): number {
        const set = new Set(nums);

        while (set.has(original)) 
            if (set.has(original)) 
                original *= 2
        
        return original
    };

    console.log(findFinalValue([5,3,6,1,12], 3))

}

File: keep-multiplying-found-values-by-two.rs
pub fn find_final_value(mut nums: Vec<i32>, mut original: i32) -> i32 {
    let n = nums.len();
    nums.sort();

    for i in 0..n {
        if nums[i] > original {
            return original;
        }

        if nums[i] == original {
            original *= 2;
        }
    }

    return original;
}
